.ci/release/Jenkinsfile

```diff
@@ -0,0 +1,235 @@
+        stage('Generate a proposed CHANGELOG') {
+          steps {
+            githubEnv()
+            generateChangelog(repo: 'apm-agent-java')
+          }
+        }
+        // 9. Go to elastic/apm-agent-java/releases and draft a new release. Provide a link to release notes in documentation as release description.
+        stage('Create GitHub release draft') {
+          steps {
+            dir("${BASE_DIR}"){
+              script {
+                def curTag = sh(script: "git tag|tail -1", returnStdout: true)
+                // Construct the URL with anchor for the release notes
+                // Ex: https://www.elastic.co/guide/en/apm/agent/java/current/release-notes-1.x.html#release-notes-1.13.0
+                def baseUrl = "https://www.elastic.co/guide/en/apm/agent/java/current/release-notes-"
+                def dotX = sh(script: 'git tag|tail -1|sed s/v//|cut -f1 -d "."|awk \'{print $1".x"}\'', returnStdout: true)
+                def bareTag = sh(script: 'git tag|tail -1|sed s/v//', returnStdout: true)
+                def finalUrl = sh(script: "echo $baseUrl$dotX\.html#release-notes-$bareTag")
+                def ret = githubReleaseCreate(draft: true, body: "[Release Notes for $bareTag]($finalUrl)")
+                env.RELEASE_ID = ret['id']
+              }
+            }
+
+          }
+        }
+        // 10. Update cloudfoundry/index.yml
+        // Needs to append a line such as the following:
+        // 1.13.0: https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent/1.13.0/elastic-apm-agent-1.13.0.jar
+        stage('Update Cloudfoundry') {
+          steps {
+            dir("${basedir}") {
+              script {
+                sh("git checkout master")
+                def bareTag = sh(script: 'git tag|tail -1|sed s/v//', returnStdout: true)
+                def baseUrl = "https://repo1.maven.org/maven2/co/elastic/apm/elastic-apm-agent"
+                def lineToAppend = sh(script: "echo '$baseUrl: $baseUrl/$bareTag/elastic-apm-agent-$bareTag\.jar' >> cloudfoundry/index.yml ")
+                gitPush()
+              }
+            }
+          }
+        }
+        // 11. Wait for released package to be available in maven central
+        stage('Wait for artifact to be available in Maven Central') {
+          steps {
+            dir("${BASE_DIR}"){
+              script {
+                // We are looking for something like this:
+                // https://oss.sonatype.org/service/local/repositories/releases/content/co/elastic/apm/apm-agent-java/1.1.12/apm-agent-java-1.1.12.pom
+                def baseUrl = "https://oss.sonatype.org/service/local/repositories/releases/content/co/elastic/apm/apm-agent-java"
+                def bareTag = sh(script: 'git tag|tail -1|sed s/v//', returnStdout: true)
+                def fullUrl = sh(script: "$baseUrl/$bareTag/apm-agent-java-$bareTag\.pom")
+                waitUntil(initialRecurrencePeriod: 15000) {
```

Comment:
Does that mean it first starts to try after 15 sec? Is there a timeout? It usually takes around 10 min.

Reply:
cURL has its own time-out, and after it expires, it will trigger the retry in the `waitUntil`. which will retry every 15 seconds, starting immediately, with a back-off. 

I think that we can improve this with the following changes:

1) Sleep first for ~10 minutes before we enter the re-try loop.
2) Bump the initial re-try up to 1 minute from 15s.

Reply:
The above strategy is now implemented. Thanks for the feedback!